sbol_utilities/component.py

```diff
@@ -572,6 +572,176 @@ def backbone(identity: str, sequence: str, dropout_location: List[int], fusion_s
+    """Digests a Component using the provided restriction enzymes and creates a product Component and a digestion Interaction.
+
+    :param reactant: DNA to be digested as SBOL Component. 
+    :param restriction_enzymes: Restriction enzymes used  Externally Defined.
+    :return: A tuple of Component and Interaction.
+    """
+    # This lists start empty but we will be filled as the algorithm proceeds
+    participations=[]
+    restriction_enzyme_names=[]
+    restriction_enzymes_pydna=[]
+    # Loop through reastriction_enzymes to make Participations.
+    for r in restriction_enzymes:
+        restriction_enzyme_names.append(r.name)
+        
+    for n in restriction_enzyme_names:
+        load = f'from Bio.Restriction import {n}'
+        save = f'restriction_enzymes_pydna.append({n})'
+        exec(load)
+        exec(save)
+ 
+    # Inform topology to PyDNA
+    if any(n==sbol3.SO_CIRCULAR for n in reactant.types):
+        circular=True
+        linear=False
+    elif any(n==sbol3.SO_LINEAR for n in reactant.types):
+        circular=False
+        linear=True
+    else:
+        circular=False
+        linear=False
+    # check that the reactant has only one DNA sequence
+    if len(reactant.sequences)==1:
+        pass
+    else: raise NotImplementedError(f'The reactant needs to have only one sequence. The input reactant has {len(reactant.sequences)} sequences')
+    reactant_seq = reactant.sequences[0].lookup().elements
+    # Dseqrecord is from PyDNA package with reactant sequence
+    ds_reactant = Dseqrecord(reactant_seq, linear=linear, circular=circular)
+    digested_reactant = ds_reactant.cut(restriction_enzymes_pydna)
+
+
+    if len(digested_reactant)==0 or len(digested_reactant)>3:
+        raise NotImplementedError(f'Not supported number of products. Found{len(digested_reactant)}')
+    elif any(n==sbol3.SO_CIRCULAR for n in reactant.types) and len(digested_reactant)==2:
+        digested_reactant = ds_reactant.cut(restriction_enzymes_pydna) #how do you extract the part if there are too many cuts?
+        # check digested_reactant
+        part_extract, backbone = digested_reactant
+    elif any(n==sbol3.SO_LINEAR for n in reactant.types) and len(digested_reactant)==3:
+        digested_reactant = ds_reactant.cut(restriction_enzymes_pydna) #how do you extract the part if there are too many cuts?
+        # check digested_reactant
+        prefix, part_extract, suffix = digested_reactant
+    else: raise NotImplementedError('The reactant has no valid topology type')
+    # Build product Component and Participation
+    reactant_roles = []
+    for f in reactant.features:
+        for r in f.roles:
+             reactant_roles.append(r)
+    # if part
+    if any(n==tyto.SO.engineered_insert for n in reactant_roles):
+        product_sequence = part_extract.seq
+        prod_comp, prod_seq = dna_component_with_sequence(identity='part_extract', sequence=str(product_sequence))
```

Comment:
I dont know which hard-coded identities should I be considering. Can you give me examples pls?

Reply:
If you run two digests (as you must in order to ligate parts), then you will try to make two components named "part_extract". This will fail because of the name collision.